runtime/parachains/src/paras/mod.rs

```diff
@@ -291,8 +291,49 @@ pub struct ParaGenesisArgs {
 	pub genesis_head: HeadData,
 	/// The initial validation code to use.
 	pub validation_code: ValidationCode,
-	/// True if parachain, false if parathread.
-	pub parachain: bool,
+	/// Parachain or Parathread.
+	pub parakind: ParaKind,
+}
+
+/// Distinguishes between Parachain and Parathread
+#[derive(PartialEq, Eq, Clone, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum ParaKind {
+	Parathread,
+	Parachain,
+}
+
+// manual encoding, decoding, adn TypeInfo as the parakind field in ParaGenesisArgs used to be a bool
+impl Encode for ParaKind {
+	fn size_hint(&self) -> usize {
+		true.size_hint()
+	}
+
+	fn using_encoded<R, F: FnOnce(&[u8]) -> R>(&self, f: F) -> R {
+		match self {
+			ParaKind::Parachain => true.using_encoded(f),
+			ParaKind::Parathread => false.using_encoded(f),
+		}
+	}
+}
+
+impl Decode for ParaKind {
+	fn decode<I: parity_scale_codec::Input>(
+		input: &mut I,
+	) -> Result<Self, parity_scale_codec::Error> {
+		match bool::decode(input) {
+			Ok(true) => Ok(ParaKind::Parachain),
+			Ok(false) => Ok(ParaKind::Parathread),
+			_ => Err("Invalid ParaKind representation".into()),
+		}
+	}
+}
+
+impl TypeInfo for ParaKind {
```

Comment:
nit: Have you considered deriving this? See e.g. https://github.com/paritytech/polkadot/blob/m-cat%2Fpvf-preparation-retry/primitives/src/v2/mod.rs#L838

Reply:
Never mind, saw the comment above.